lib/process.js

```diff
@@ -10,12 +10,34 @@ exports.processCommits = function processCommits(options, commits, range) {
     return Promise.reject(new Error('No commits in the specified range'));
   }
 
-  var externalScriptPath = options.s || options.script;
-	if (!externalScriptPath) {
+	var debugScript = require("debug")("release-notes:externalscript");
+	var funcOrPath = options.s || options.script;
+	if (!funcOrPath) {
     debug("Rendering template without post processing");
     return Promise.resolve({ commits: commits });
-  }
+	}
+
+	if (typeof funcOrPath === 'function') {
+		var externalFunction = funcOrPath;
+
+		return new Promise((resolve, reject) => {
+			try {
+				externalFunction({
```

Comment:
The code is good, one minor comment, can you move this object up?

```js
var inputData = {
  commits: commits,
	range: range,
	dateFnsFormat: dateFnsFormat,
	debug: debugScript
}
```

because it's duplicate few lines below, I want to be sure the both the script and the function will receive the same inputs.